### DISCUSSION TITLE
What are the variables for in the csv file generated in a simulation?

### DISCUSSION THREAD

**Discussion by pshen20127** [idaholab/moose#26287] [Q&A Modules: Solid mechanics]
Hi All

After I finished a simulation, I got 2 files as below. 
![Weixin Screenshot_20231208204809](https://github.com/idaholab/moose/assets/81499506/a0b25e93-7ed5-4cc8-b827-2c7e646fff04)
When I open the .csv file, there are several variables as shown in the red block. Which node or point are these values for?
![Weixin Screenshot_20231208205119](https://github.com/idaholab/moose/assets/81499506/da662228-36b5-4ea3-846d-5478c1bdfb98)

Thanks
Pengfei

**Comment by GiudGiud** [comment]
Hello

The content of the CSV file depends on the input file.
The typical case is that it contains the output of postprocessors and scalar variables on every time step.
Please check the `[Postprocessors]` block in your input file.

Guillaume

**Reply by pshen20127** [reply]
Hello

I know they are output of postprocessors and scalar variables on every time step. I am just wondering are these values for a node or element? if so, which node or element are these values for? Because it outputs euler angles, which must be a certain element's euler angle, I want to know which element it refers to. I attached my input file and output .csv file in the link below.
https://github.com/pshen20127/Test.git

Pengfei

**Reply by GiudGiud** [reply]
Looking at the Postprocessor block on the input on the repo, these are averages of the element values. 

Or more accurately averages over the entire mesh using an integration loop overs element (integrates over quadrature points on each element)

**Reply by pshen20127** [reply]
Then are the euler angles the averages of all elements? Another question is how I get engineering stress and strain and plot them?

Pengfei

**Reply by GiudGiud** [reply]
These ones?
```
  [euler_angle_1]
    type = ElementAverageValue
    variable = euler_angle_1
  []
  [euler_angle_2]
    type = ElementAverageValue
    variable = euler_angle_2
  []
  [euler_angle_3]
    type = ElementAverageValue
    variable = euler_angle_3
  []
```

yes they are averages over all elements.
I think you can get stresses & strains out from material property output.
There's been questions about getting the engineering stress on the forum already i think as well
